hibernate-core/src/main/java/org/hibernate/MultiIdentifierLoadAccess.java

```diff
@@ -69,6 +69,31 @@
 	 */
 	MultiIdentifierLoadAccess<T> enableSessionCheck(boolean enabled);
 
+	/**
+	 * Specify whether we should check the {@link SessionFactory} to see whether the second-level cache already contains any of the
+	 * entities to be loaded <b>for the purpose of not including those
+	 * ids to the batch-load SQL</b>.
+	 *
+	 * @param enabled {@code true} enables this checking; {@code false} (the default) disables it.
+	 *
+	 * @return {@code this}, for method chaining
+	 */
+	MultiIdentifierLoadAccess<T> enableSessionFactoryCheck(boolean enabled);
```

Comment:
How is this different than `MultiIdentifierLoadAccess<T> with(CacheMode cacheMode)`, which we already have?  This should just go away imo